samples/tutorial/Tutorial4.md

```diff
@@ -0,0 +1,343 @@
+    props: TodoProps,
+    state: State,
+    context: RenderContext
+  ): List<Any> {
+    val todoListScreen = context.renderChild(
+        TodoListWorkflow,
+        props = ListProps(
+            name = props.name,
+            todos = state.todos
+        )
+    ) { output ->
+      when (output) {
+        Output.Back -> onBack()
+        is SelectTodo -> editTodo(output.index)
+        NewTodo -> newTodo()
+      }
+    }
+
+    return listOf(todoListScreen)
+  }
+
+  private fun onBack() = action {
+    // When an onBack action is received, emit a Back output.
+    setOutput(Back)
+  }
+
+  private fun editTodo(index: Int) = action {
+    // When a todo item is selected, edit it.
+    state = state.copy(step = Step.Edit(index))
+  }
+
+  private fun newTodo() = action {
+    // Append a new todo model to the end of the list.
+    state = state.copy(
+        todos = state.todos + TodoModel(
+            title = "New Todo",
+            note = ""
+        )
+    )
+  }
+}
+```
+
+Update the `RootWorkflow` to defer to the `TodoWorkflow` for rendering the `Todo` state. This will get us back into a state where we can build again (albeit without editing support):
+
+```kotlin
+object RootWorkflow : StatefulWorkflow<Unit, State, Nothing, BackStackScreen<*>>() {
+
+  // …
+
+  override fun render(
+    props: Unit,
+    state: State,
+    context: RenderContext
+  ): BackStackScreen<*> {
+
+    // Our list of back stack items. Will always include the "WelcomeScreen".
+    val backstackScreens = mutableListOf<Any>()
+
+    // Render a child workflow of type WelcomeWorkflow. When renderChild is called, the
+    // infrastructure will create a child workflow with state if one is not already running.
+    val welcomeScreen = context.renderChild(WelcomeWorkflow) { output ->
+      // When WelcomeWorkflow emits LoggedIn, turn it into our login action.
+      login(output.name)
+    }
+    backstackScreens += welcomeScreen
+
+    when (state) {
+      // When the state is Welcome, defer to the WelcomeWorkflow.
+      is Welcome -> {
+        // We always add the welcome screen to the backstack, so this is a no op.
+      }
+
+      // When the state is Todo, defer to the TodoListWorkflow.
+      is Todo -> {
+        val todoListScreens = context.renderChild(TodoWorkflow, TodoProps(state.name)) {
+          // When receiving a Back output, treat it as a logout action.
+          logout()
+        }
+        backstackScreens.addAll(todoListScreens)
+      }
+    }
+
+    // Finally, return the BackStackScreen with a list of BackStackScreen.Items
+    return backstackScreens.toBackStackScreen()
+  }
+
+  // …
+
+}
+```
+
+#### Moving Edit Output handling to the TodoWorkflow
+
+The `TodoWorkflow` now can handle the outputs from the `TodoListWorkflow`. Next, let's add handling for the `TodoEditWorkflow` output events.
+
+Since the types of output and actions are pretty different from their origin, make a *second* action type on the `TodoWorkflow`:
```

Comment:
We don't acually have action types in this example, do we?

Reply:
Nope. Changing this to "*second* set of actions"